crates/egui-wgpu/src/winit.rs

```diff
@@ -691,6 +684,13 @@ impl Painter {
             vsync_sec += start.elapsed().as_secs_f32();
         };
 
+        {
```

Comment:
```suggestion
        // Free textures marked for destruction **after** queue submit since they might still be used in the current frame.
        // Calling `wgpu::Texture::destroy` on a texture that is still in use would invalidate the command buffer(s) it is used in.
        // However, once we called `wgpu::Queue::submit`, it is up for wgpu to determine how long the underlying gpu resource has to live.
        {
```